### PR TITLE
[Add] Add blocksparse_gemm and grouped_gemm_bwd operators with comprehensive design documentation

### PR DESCRIPTION
**Pull request by handsomeRobotSK**
## Overview

This PR introduces two new TileLang-Ascend operator implementations, along with complete design documentation and test validation.

### 1. BlockSparse GEMM
- **Functionality**: Block-sparse matrix multiplication with configurable BlockMask to control sparsity
- **Implementation**: `examples/blocksparse_gemm/example_blocksparse_gemm.py`
- **Design Document**: `examples/blocksparse_gemm/design.md`
- **Test Coverage**:
  - ✅ Level 0: Basic functionality validation
  - ✅ Level 1: Typical configuration validation
  - ✅ Level 2: Edge cases (fully dense + extremely sparse)
  - ✅ Level 3: Auto-tuning support

### 2. Grouped GEMM Backward
- **Functionality**: Grouped matrix multiplication backward pass (weight gradient computation)
- **Implementation**: `examples/grouped_gemm/example_grouped_gemm_bwd.py`
- **Design Document**: `examples/grouped_gemm/design.md`
- **Features**:
  - Supports dynamic batch sizes
  - Uses `block_metadata` mechanism to handle non-uniform batches
  - Developer mode implementation

## Documentation Updates

- Updated the `tilelang-op-design` skill template to enhance technical constraint detection
- Updated `AGENTS.md` to refine operator development process instructions

## Example Modifications

- Added explicit `target="ascendc"` to `example_gemm_intrinsic.py` and `example_gemm_intrinsic_persistent.py`

## Test Results

All new operators have passed validation:
- **BlockSparse GEMM**: All test levels passed (basic, typical, edge, auto-tuning)
- **Grouped GEMM Backward**: Correctness aligns with reference implementation

## Dependency Checks

- [x] No new external dependencies introduced
- [x] Follows project code style guidelines
- [x] Design documents conform to template requirements

### CODE REVIEW

File: .agents/skills/tilelang-op-design/SKILL.md [hunk 1]
```diff
@@ -109,47 +189,85 @@ description: "根据算子需求生成 TileLang-Ascend 算子设计文档（desi
...
+### 4.0 平台识别
+
+**本项目为 TileLang-Ascend（昇腾 NPU）**，与 GPU 版 TileLang 有差异：
+
+| 差异项 | GPU 版 TileLang | 本项目（Ascend） |
+|--------|----------------|-----------------|
+| Kernel 维度 | 支持三维 | **只支持一维** |
+| threads 参数 | 支持 128+ | **只支持 1 或 2** |
+| 循环边界 | 支持动态 | **只支持静态** |
+| GEMM API | `T.gemm` | **`T.gemm_v0` / `T.gemm_v1`** |
```
Comment (LLMZhangYC) [inline .agents/skills/tilelang-op-design/SKILL.md:204]: gemm_v1现在可以用吗，确认下

**Review by fuhouyu-hw — DISMISSED** [review]

File: examples/grouped_gemm/example_grouped_gemm_bwd.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+import tilelang.language as T
+
+tilelang.cache.clear_cache()
+
+
+def torch_grouped_gemm_bwd(A, B, block_metadata):
+    """
+    Golden function: compute dB_i = A_i^T @ B_i for each group
+    """
+    batch_count = A.shape[1]  # M dimension
```
Comment (LLMZhangYC) [inline examples/grouped_gemm/example_grouped_gemm_bwd.py]: Here batch_count == M, check if it is correct.

File: examples/grouped_gemm/example_grouped_gemm_bwd.py [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+        m_block_idx = int(row[1])
+        n_block_idx = int(row[2])
+        batch_offset = int(row[3])
+        k_iters = int(row[4])
+
+        # Skip invalid blocks
+        if k_iters == 0:
+            continue
+
+        m_start = m_block_idx * 64
```
Comment (LLMZhangYC) [inline examples/grouped_gemm/example_grouped_gemm_bwd.py]: 这里*64是block_M/block_N的大小吗

File: examples/blocksparse_gemm/example_blocksparse_gemm.py [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_COMBINE: True,
+    tilelang.PassConfigKey.TL_ASCEND_AUTO_CV_SYNC: True,
+}
+
+@tilelang.autotune(
+    configs=get_configs(),
+)
+@tilelang.jit(out_idx=[-1], pass_configs=pass_configs)
+def blocksparse_matmul(
+    M, N, K, block_M, block_N, block_K, num_stages, dtype="float16", accum_dtype="float"
```
Comment (LLMZhangYC) [inline examples/blocksparse_gemm/example_blocksparse_gemm.py]: Useless variable "num_stages".

File: examples/blocksparse_gemm/example_blocksparse_gemm.py [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+):
+    m_num = M // block_M
```
Comment (LLMZhangYC) [inline examples/blocksparse_gemm/example_blocksparse_gemm.py:63]: Do M/N need extra process when cannot be divided by block_M/block_N?

File: .agents/skills/tilelang-op-design/SKILL.md (hunk 1 above)
Comment (handsomeRobotSK) [reply .agents/skills/tilelang-op-design/SKILL.md:204]: 当前暂时不可用。不过，skill 会自动参考其他算子的使用方式，此处标记可为后续 gemm_v1 支持时实现平滑过渡。是否需要先删除该标记？

File: examples/grouped_gemm/example_grouped_gemm_bwd.py (hunk 2 above)
Comment (handsomeRobotSK) [reply examples/grouped_gemm/example_grouped_gemm_bwd.py]: batch_count改为参数传入

File: examples/grouped_gemm/example_grouped_gemm_bwd.py (hunk 3 above)
Comment (handsomeRobotSK) [reply examples/grouped_gemm/example_grouped_gemm_bwd.py]: 是的，已修改硬编码部分

File: examples/blocksparse_gemm/example_blocksparse_gemm.py (hunk 4 above)
Comment (handsomeRobotSK) [reply examples/blocksparse_gemm/example_blocksparse_gemm.py]: 无效变量已删除

File: .agents/skills/tilelang-op-design/SKILL.md (hunk 1 above)
Comment (LLMZhangYC) [reply .agents/skills/tilelang-op-design/SKILL.md:204]: 先删除吧

**Review by fuhouyu-hw — DISMISSED** [review]

**Review by LLMZhangYC — APPROVED** [review]
lgtm

**Review by fuhouyu-hw — APPROVED** [review]
